modules/shortcodes/instagram.php

```diff
@@ -177,47 +220,50 @@ function jetpack_instagram_handler( $matches, $atts, $url ) {
 	 * @param array  $atts       Instagram Shortcode attributes.
 	 * @param string $passed_url Instagram API URL.
 	 */
-	$response_body_use_cache = apply_filters( 'instagram_cache_oembed_api_response_body', false, $matches, $atts, $passed_url );
-	$response_body           = false;
-	if ( $response_body_use_cache ) {
-		$cache_key     = 'oembed_response_body_' . md5( $url );
-		$response_body = wp_cache_get( $cache_key, 'instagram_embeds' );
-	}
+	return apply_filters( 'instagram_cache_oembed_api_response_body', false, $matches, $atts, $passed_url );
+}
 
-	if ( ! $response_body ) {
-		// Not using cache (default case) or cache miss.
-		$instagram_response = wp_remote_get( $url, array( 'redirection' => 0 ) );
-		if (
-			is_wp_error( $instagram_response )
-			|| 200 !== $instagram_response['response']['code']
-			|| empty( $instagram_response['body'] ) ) {
-			return '<!-- instagram error: invalid instagram resource -->';
+/**
+ * Handles the logic for actually fetching information for an Instagram embed, conditionally proxying the
+ * request through the WordPress.com API if this is not a WordPress.com site.
+ *
+ * @param array $args An array of arguments to pass to the embed API.
+ *
+ * @return mixed A HTML comment (string) if errored or an object.
+ */
+function jetpack_instagram_fetch_embed( $args ) {
+	if ( Constants::is_defined( 'IS_WPCOM' ) && Constants::get_constant( 'IS_WPCOM' ) ) {
+		$url      = esc_url_raw(
+			add_query_arg(
+				$args,
+				'https://api.instagram.com/oembed/'
+			)
```

Comment:
Maybe we should pass that full URL into the `jetpack_instagram_fetch_embed` function, so we don't have to create that `$url` twice?

Reply:
Normally, I'd agree. But, we'll need to refactor this slightly for thee October release to 1) point to https://graph.facebook.com/v5.0/instagram_oembed? and 2) allow specifying a token via a filter (WPCOM, VIP, etc.). So, in this case, it's short-lived.